src/compiler/generate-common-tests.xsl

```diff
@@ -168,7 +188,7 @@
        corresponding call instruction at some point).
    -->
    <xsl:template name="x:call-scenarios">
-      <xsl:param name="pending" select="this//@focus" tunnel="yes" as="node()?"/>
+      <xsl:param name="pending" select="(.//@focus)[1]" tunnel="yes" as="node()?"/>
```

Comment:
Here is what I think I'm seeing in the XSLT case. I haven't looked at the XQuery case as much.

When the `name="x:call-scenarios"` template operates on `x:description`, the default value is assigned to the `$pending` tunnel parameter. However, the value does not matter because the generated `name="t:main"` template needs to call the generated templates corresponding to all the top-level scenarios, whether they are pending or not. (Pending status affects deeper-level parts of the generated code.)

When the `name="x:call-scenarios"` template operates on `x:scenario`, the default value of the `$pending` tunnel parameter does not get assigned because a value came from the calling `name="x:output-scenario"` template. Before that, the `$pending` value came from the calling `match="x:scenario" mode="x:compile"` template's `$new-pending` variable, which reflects whether the scenario being operated on should be treated as pending.

Please correct anything that sounds inaccurate. I want to understand the compiler code and it's complicated.

I agree with you that the default does not matter in the current code path. As for possible future changes, I'm not sure whether `select="(.//@focus)[1]"` or `select="()"` is more suitable; maybe whoever makes the future changes will need to think through the initialization and propagation of `$pending` based on what those changes are trying to do. I do think that what you have makes more sense than a reference to a `this` element that does not exist.

Is there value in adding a comment like, "Default value of pending does not affect compiler output but is here if needed in the future"? That could save someone time trying to figure out what you now know. If you think it's extraneous, it's fine not to add it.

Reply:
Thanks for the invaluable review. Adding a comment is a very nice idea! I added it to this pull request: a98f70dffa97c61d65c013ae04e42eb3c7662e51

Reply:
You're very welcome! Thanks for adding the comment.